### ISSUE TITLE
C++: Add standalone user-defined archetype+component example

### ISSUE THREAD

**Issue by teh-cmc** [rerun-io/rerun#3139]
Once the new `as_components()`-based API land, we'll need an example to showcase how one can:
- Define a custom archetype
- Define a custom component
- Extend an existing archetype

For example: implement a `user.MyPoints3D` archetype which "embeds" a `rerun.archetypes.Points3D` archetype and augments it with a custom `user.Confidence` component.

- [x] Python
- [x] Rust
- [ ] C++

**Comment by emilk** [comment]
A lot of this is already done in `docs/code-examples/custom_data.py`

**Comment by teh-cmc** [comment]
> what is missing?

C++

**Comment by Wumpf** [comment]
We're done for 0.9 here in that case. C++ is part of 0.10
